packages/opentelemetry/src/utils/parseSpanDescription.ts

```diff
@@ -240,3 +262,17 @@ export function getSanitizedUrl(
 
   return { urlPath: undefined, url, query, fragment, hasRoute: false };
 }
+
+/**
+ * Because Otel decided to mutate span names via `span.updateName`, the only way to ensure
+ * that a user-set span name is preserved is to store it as a tmp attribute on the span.
+ * We delete this attribute once we're done with it when preparing the event envelope.
+ * @internal exported only for testing
+ */
+export function getOriginalName(name: string, attributes: Attributes): string {
+  return attributes[SEMANTIC_ATTRIBUTE_SENTRY_SOURCE] === 'custom' &&
```

Comment:
l: Do we even need to check the source here? Is it not enough to just check `_sentry_span_name_set_by_user`? My logic here: Imagine a user does this:

```js
Sentry.updateSpanName(span, 'new name');
Sentry.setAttribute('sentry.source', 'route');
```

Would we then _not_ want the name to be `new name` anymore? IMHO this guard is not necessary, if it was explicitly set, we just use it...?

Reply:
hmm good catch, thanks! I'll change it and only check for the existence of `_sentry_span_name_set_by_user`.

Reply:
I adjusted the logic in parseSpanDescription so that `_sentry_span_name_set_by_user` always takes precedence and we use the already defined source.